EpsilonDriverControls/Tests/tests/DriverControlsTest/DriverControlsTest.c

```diff
@@ -0,0 +1,926 @@
+#include "unity.h"
+
+#include "Mockstm32f4xx_hal_gpio.h"
+#include "Mockstm32f4xx_hal_can.h"
+#include "Mockstm32f4xx_hal_adc.h"
+#include "Mockcmsis_os.h"
+
+#include "DriverControls.h"
+#include "DriverControlsTest.h"
+
+ADC_HandleTypeDef hadc1;
+ADC_HandleTypeDef hadc2;
+
+osPoolId canPool;
+osMessageQId canQueue;
+
+uint8_t auxBmsInputs[2];
+float   motor0VehicleVelocityInput;
+float   motor1VehicleVelocityInput;
+
+void checkDriveMsgStdidAndDlcForDriveCommands(CanMsg* expectedMotorDriveMsg);
+void checkPowerMessageForDriveCommands(CanMsg* expectedMotorPowerMsg);
+void checkExpectMotorDriveMessage(CanMsg* expectedMotorDriveMsg);
+void checkExpectMotorPowerMessage(CanMsg* expectedMotorPowerMsg);
+void setForwardReverseAndBrakePins(uint8_t setPins[]);
+
+void runDriverControlTests()
+{
+    RUN_TEST(test_sendHeartbeat);
+    RUN_TEST(test_sendLights);
+    RUN_TEST(test_sendMusic);
+    RUN_TEST(test_sendDriver);
+    RUN_TEST(test_sendCan_EqualToHalOk);
+    RUN_TEST(test_sendCan_NotEqualToHalOk);
+    RUN_TEST(test_sendDriveCommands_wasPreviouslyResetAndWantToResetAgain);
+    RUN_TEST(test_sendDriveCommands_wasNotPreviouslyResetAndWantToReset);
+    RUN_TEST(test_sendDriveCommands_noResetsAtAll);
+    RUN_TEST(test_sendDriveCommands_forwardAndReverseIsAnErrorState);
+    RUN_TEST(test_sendDriveCommands_forwardAndMotorVehicleVelocityIsNotSafe);
+    RUN_TEST(test_sendDriveCommands_reverseAndMotorVehicleVelocityIsNotSafe);
+    RUN_TEST(test_sendDriveCommands_regenPercentageGreaterThanNonZeroThresholdAndAllowingCharge);
+    //putting this above regenPercentageGreaterThanNonZeroThresholdAndAllowChargeIsFalse will cause it to not break if the HAL ADC is commented
```

Comment:
I think this comment isn't correct anymore, I moved this function around and couldn't get anything to break?